// Filename: addressbook.proto

syntax="proto2";
package book;

//import "src/help.proto";      //举例用，编译时去掉

message Person {
    required string name = 1;
    required int32 id = 2;
    optional string email = 3;

    enum PhoneType {
        MOBILE = 0;
        HOME = 1;
        WORK = 2;
    }

    message PhoneNumber {
        optional string number = 1;
        optional PhoneType type = 2 ;

        repeated Address address = 3;
    }

    repeated PhoneNumber phone = 4;
}

message Address{
    optional string street = 1;
    optional int32 house_number = 2;
}

message AddressBook {
    repeated Person person_info = 1;
}